ogr/ogrsf_frmts/hana/ogrhanatablelayer.cpp

```diff
@@ -0,0 +1,1762 @@
+        return (typeInfo.width == 0)
+                ? typeInfo.name
+                : StringFormat("%s(%d)", typeInfo.name.c_str(), typeInfo.width);
+    default:
+        return "UNKNOWN";
+    }
+}
+
+void SetFieldDefn(OGRFieldDefn& field, const ColumnTypeInfo& typeInfo)
+{
+    auto isArray = [&typeInfo]() {
+        return std::strstr(typeInfo.name, "ARRAY") != nullptr;
+    };
+
+    switch (typeInfo.type)
+    {
+    case odbc::SQLDataTypes::Bit:
+    case odbc::SQLDataTypes::Boolean:
+        field.SetType(OFTInteger);
+        field.SetSubType(OFSTBoolean);
+        break;
+    case odbc::SQLDataTypes::TinyInt:
+    case odbc::SQLDataTypes::SmallInt:
+        field.SetType(isArray() ? OFTIntegerList : OFTInteger);
+        field.SetSubType(OFSTInt16);
+        break;
+    case odbc::SQLDataTypes::Integer:
+        field.SetType(isArray() ? OFTIntegerList : OFTInteger);
+        break;
+    case odbc::SQLDataTypes::BigInt:
+        field.SetType(isArray() ? OFTInteger64List : OFTInteger64);
+        break;
+    case odbc::SQLDataTypes::Double:
+    case odbc::SQLDataTypes::Real:
+    case odbc::SQLDataTypes::Float:
+        field.SetType(isArray() ? OFTRealList : OFTReal);
+        if (typeInfo.type != odbc::SQLDataTypes::Double)
+            field.SetSubType(OFSTFloat32);
+        break;
+    case odbc::SQLDataTypes::Decimal:
+    case odbc::SQLDataTypes::Numeric:
+        field.SetType(isArray() ? OFTRealList : OFTReal);
+        break;
+    case odbc::SQLDataTypes::Char:
+    case odbc::SQLDataTypes::VarChar:
+    case odbc::SQLDataTypes::LongVarChar:
+        field.SetType(isArray() ? OFTStringList : OFTString);
+        break;
+    case odbc::SQLDataTypes::WChar:
+    case odbc::SQLDataTypes::WVarChar:
+    case odbc::SQLDataTypes::WLongVarChar:
+        field.SetType(isArray() ? OFTStringList : OFTString);
+        break;
+    case odbc::SQLDataTypes::Date:
+    case odbc::SQLDataTypes::TypeDate:
+        field.SetType(OFTDate);
+        break;
+    case odbc::SQLDataTypes::Time:
+    case odbc::SQLDataTypes::TypeTime:
+        field.SetType(OFTTime);
+        break;
+    case odbc::SQLDataTypes::Timestamp:
+    case odbc::SQLDataTypes::TypeTimestamp:
+        field.SetType(OFTDateTime);
+        break;
+    case odbc::SQLDataTypes::Binary:
+    case odbc::SQLDataTypes::VarBinary:
+    case odbc::SQLDataTypes::LongVarBinary:
+        field.SetType(OFTBinary);
+        break;
+    default:
+        break;
+    }
+
+    field.SetWidth(typeInfo.width);
+    field.SetPrecision(typeInfo.precision);
+}
+
+} // namespace
+
+/************************************************************************/
+/*                         OGRHanaTableLayer()                          */
+/************************************************************************/
+
+OGRHanaTableLayer::OGRHanaTableLayer(OGRHanaDataSource* datasource, int update)
+    : OGRHanaLayer(datasource)
+    , updateMode_(update)
+    , batchSize_(4 * 1024)
```

Comment:
I believe the doc says the default batch size is 1 KB not 4 KB

Reply:
Good catch. The value was changed to 4MB in d14c72057258326c1e697eb32715f605ba622237.